internal/brclient/brclient.go

```diff
@@ -1,15 +1,49 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package brclient
 
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/gardener/etcd-wrapper/internal/types"
+	"github.com/gardener/etcd-wrapper/internal/util"
+)
+
 // InitStatus is the status of initialisation as returned from backup-restore.
 type InitStatus int
 
 const (
+	// Unknown indicates that the initialisation by backup-restore is unknown.
+	Unknown InitStatus = iota
+	// New indicates that the initialisation by backup-restore is new and has not started yet.
+	New
 	// InProgress indicates that the initialisation by backup-restore is in-progress.
-	InProgress InitStatus = iota
-	// Success indicates that the initialisation by backup-restore is successful.
-	Success
+	InProgress
+	// Successful indicates that the initialisation by backup-restore is successful.
+	Successful
```

Comment:
where's the initStatus `Failed` ? are you guys not using it or `not successful` considered as `Failed` ?

Reply:
Yes, exactly. Any status other than `Successful` means that we retry

Reply:
you can also check the existing handling of Failed status (https://github.com/gardener/etcd-custom-image/blob/master/etcd_bootstrap_script.sh#L54). Essentially just again retry.